.gitignore

```diff
@@ -9,3 +9,15 @@
 
 # OSX
 .DS_Store
+
+# Editor-specific settings
+.clang_complete
+
+# CMake files
```

Comment:
I think I misunderstood your question from yesterday. If you run cmake from the build directory (i.e. `cd build; cmake ..`), then all of the files should be covered by the .gitignore in the build directory.